### ISSUE TITLE
Not working on Ubuntu 19.10 / Snap Spotify / hyper-statusline

### ISSUE THREAD

**Issue by Bonlou** [panz3r/hyper-spotify#29]
On a almost-fresh install of Ubuntu 19.10, with a Spotify installed via `snap`, I have just installed Hyper and added a few plugins:
```javascript
plugins: [
  'hyper-spotify',
  'hyper-material-theme',
  'hyper-statusline',
  'hyper-tab-icons',
  'hyper-dark-scrollbar'
],
```
Sadly, it seems to not working, even with `hyper-statusline` disabled.
I just have a Spotify icon in the middle of the bar, which I can't click when `hyper-statusline` is installed. If I remove it, and then full reload, the Spotify icon just open a new instance of Spotify, ignoring the one running, and if I play music in this one, the control icons are still not displayed.

Is it supposed to work the way I'm using it ?

Hope it will help future development, thanks for your work !

**Comment by panz3r** [comment]
Hi @Bonlou ,

Thanks for your feedback.

Regarding the layout issues when using together with `hyper-statusline` there's a setting for `hyper-spotify` to fix the positioning issues (I designed this plugin to work alongside `hyper-statusline` since I'm also a great fan of that 😬).
Inside your `~/.hyper.js` config file add this
```js
modules.exports = {
  config: {
    ...
    hyperSpotify: {
      margin: 'double',
    },
    ...
  },
  ... 
};
```

 I'll look into the problem related to the connection with the running Spotify.

**Comment by Bonlou** [comment]
Hi, thanks for your answer !
Sadly, this config seems not to work.


Actually, the Spotify icon is only visible if it is before statusline in the plugin array.

![image](https://user-images.githubusercontent.com/29218400/69006611-9d942280-0931-11ea-9f28-703d3b464a87.png)


When it is, it disappear when hover, as shown here :

![Peek 17-11-2019 12-01](https://user-images.githubusercontent.com/29218400/69006658-6ffba900-0932-11ea-8525-84257c2019ae.gif)


When inspecting Hyper and enable "pause on exceptions", it will stop here:

![image](https://user-images.githubusercontent.com/29218400/69006600-72113800-0931-11ea-8dbc-1cfb13af2d68.png)
Hope these feedback will help you investigate, good luck !

**Comment by panz3r** [comment]
Hi @Bonlou ,

Thanks again for the details.

Regarding the `hyperSpotify` config, that must be put inside the `config` section of your `~/.hyper.js` config file.
To be more clear under the
```js
    // for advanced config flags please refer to https://hyper.is/#cfg
```
comment part, inside that section (before the `}`)

```js
...
    // rendering (slower, but supports transparent backgrounds)
    webGLRenderer: true,

    // for advanced config flags please refer to https://hyper.is/#cfg

    hyperSpotify: {
      margin: 'double',
    },
  },

  // a list of plugins to fetch and install from npm
  // format: [@org/]project[#version]
  // examples:
  //   `hyperpower`
  //   `@company/project`
  //   `project#1.0.1`
  plugins: [
    'hyper-spotify',
    ...
  ],
....
```

I'm working on the error you reported about the uncaught exception, there seems to be an issue with the module I'm currently using to connect to `Spotify` on Linux.

I hope to have a working version of the new code by tomorrow.

**Comment by Bonlou** [comment]
Oh, didn't notice the `config` object wasn't whole part of the `.hyper.js` file, thanks for pointing out !
Thanks for your involvement !

**Comment by panz3r** [comment]
Hi @Bonlou ,

Thanks again for your feedback, I just released a new version in test mode and I'd be glad if you could test it out and see if it solves your issue (I made a quick test and it seems to work just fine).

To install the test version: 
- change the plugin to `hyper-spotify@next` inside `~/.hyper.js` and save
- click on `Plugins > Update' inside Hyper menu
- when Hyper ask you to 'Reload' do a 'Full Reload' from Hyper menu

Let me know if this works as expected.

**Comment by Bonlou** [comment]
Hi @panz3r !
It work like a charm, but only if the plugin is declared _before_ `hyper-statusline`. Else, it is successfully loaded but behave like if it was not.
Anyway, thanks for the quick fix !!

**Comment by panz3r** [comment]
Hi @Bonlou,

Thanks for your testing, glad to hear it works 😄 
I just released the `5.3.1` version as stable so you can remove the `@next` tag from the plugin.
